site/content/docs/main/tilt.md

```diff
@@ -0,0 +1,176 @@
+```
+spec:
+  config:
+    region: minio
+    s3ForcePathStyle: "true"
+    s3Url: http://minio.velero.svc:9000
+  objectStorage:
+    bucket: velero
+  provider: aws
+```
+
+Here are two ways to use MinIO as the storage:
+
+1) As a MinIO instance running inside your cluster
+
+    In the `tilt-settings.json` file, set `"setup-minio": true`. This will configure a Kubernetes deployment containing a running
+instance of Minio inside your cluster. There are [extra steps](https://velero.io/docs/main/contributions/minio/#expose-minio-outside-your-cluster-with-a-service)
+necessary to expose Minio outside the cluster. Note: with this setup, when your cluster is terminated so is the storage and any backup/restore in it.
+
+2) As a standalone MinIO instance running locally in a Docker container
+
+    See [these instructions](https://github.com/vmware-tanzu/velero/wiki/Contributing-FAQ#minio) to run MinIO locally on your computer, as a standalone as opposed to running it on a Pod.
+
+Please see our [locations documentation](https://velero.io/docs/main/locations/) to learn more how backup locations work.
+
+### Configure the provider credentials (secret)
+Whatever object storage provider you use, configure the credentials for in the `velero/tilt-resources/cloud` file. Read the [plugin documentation](https://velero.io/plugins/)
+to learn what field/value pairs are required for your provider's credentials. The Tilt file will invoke Kustomize to create the secret under the hard-coded key `secret.cloud-credentials.data.cloud` in the Velero namespace.
+
+There is a sample credentials file properly formatted for a MinIO storage credentials in `velero/tilt-resources/examples/cloud`.
+
+### Run Tilt!
+To launch your development environment, run:
+
+``` bash
+tilt up
+```
+
+This will output the address to a web browser interface where you can monitor Tilt's status and the logs for each Tilt resource. After a brief amount of time, you should have a running development environment, and you should now be able to
+create backups/restores and fully operate Velero. 
+
+Note: Running `tilt down` after exiting out of Tilt [will delete all resources](https://docs.tilt.dev/cli/tilt_down.html) specified in the Tiltfile.
+
+Tip: Create an alias to `velero/_tuiltbuild/local/velero` and you won't have to run `make local` to get a refreshed version of the Velero CLI, just use the alias.
+
+Please see the documentation for [how Velero works](https://velero.io/docs/main/how-velero-works/).
+
+## Provider plugins
+A provider must supply a `tilt-provider.json` file describing how to build it. Here is an example:
+
+```json
+{
+  "plugin_name": "velero-plugin-for-aws",
+  "context": ".",
+  "image": "velero/velero-plugin-for-aws",
+  "live_reload_deps": [
+    "velero-plugin-for-aws"
+  ],
+  "go_main": "./velero-plugin-for-aws"
+}
+```
+
+## Live updates
```

Comment:
Will this also watch and do a live reload of the plugin images?

Reply:
How about updates to the restic-restore-helper image?

Reply:
> Will this also watch and do a live reload of the plugin images?

This is what happens with the plugins, as documented:
> When a dependency is modified, Tilt rebuilds the provider's binary **on your local
machine**, copies the binary to the init container, and triggers a restart of the Velero container.

Reply:
> How about updates to the restic-restore-helper image?

Going to address this not in this PR.